### PR TITLE
Karate - Rest API Test Automation Framework for audit api tests

### PR DESCRIPTION
**Pull request by kumarvarun1252**
- Added a project for Rest API test automation
- Uses Karate framework
- Added some example tests for api-audit service
- Same can be used for api service and other services as well

### CODE REVIEW

File: api-tests/.gitignore [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+.idea
+*.iml
+
+# Cucumber output
+target
+
+# Node
+node_modules
+
+package-lock.json
```
Comment (amit1s) [inline api-tests/.gitignore]: Add newline to remove warning

File: api-tests/README.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+## Setup and Run Tests
+
+Make sure a local instance of [api-audit service](http://capitalone.github.io/Hygieia/api-audit.html) is running on your local host. 
+
+If you running tests for api-audit
+
+```$> cd Hygieia/api-tests/api-audit```
+
+```$> mvn clean install```
```
Comment (amit1s) [inline api-tests/README.md]: Do we want to explicitly call out how to run individual tests?
"mvn test" to run all of them 
"mvn test -Dcucumber.options=..."
Specially for peerReview or dashboardReview

File: api-tests/api-audit/src/test/java/basic-auth.js [hunk 3]
```diff
@@ -0,0 +1,9 @@
+
+function basicAuth() {
+    //apiToken needs to be updated accordingly
```
Comment (amit1s) [inline api-tests/api-audit/src/test/java/basic-auth.js]: Use .properties for apitoken configuration

File: api-tests/api-audit/src/test/java/endpoints/codeReview/peerReview.feature [hunk 4]
```diff
@@ -0,0 +1,14 @@
+Feature: peerReview
+
+  Background:
+    * header apiUser = 'apiadmin'
```
Comment (amit1s) [inline api-tests/api-audit/src/test/java/endpoints/codeReview/peerReview.feature]: Consider adding apiadmin to the .properties file suggestion in basic-auth.js

File: api-tests/api-audit/src/test/java/endpoints/codeReview/peerReview.feature [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+    * header Authorization = call read('classpath:basic-auth.js')
+    * url baseUrl
+    * configure readTimeout = 60000
+
+  Scenario: Verify pull request information for a repo
```
Comment (amit1s) [inline api-tests/api-audit/src/test/java/endpoints/codeReview/peerReview.feature]: We have more audit statuses so please add a //ToDo indicating that this is just the basic setup check
and a pointer to where those statuses are documented - for example the readme in api-audit

File: api-tests/api-audit/src/test/java/endpoints/dashboard/dashboardReview.feature [hunk 6]
```diff
@@ -0,0 +1,15 @@
+Feature: dashboardReview
+
+  Background:
+    * header apiUser = 'apiadmin'
```
Comment (amit1s) [inline api-tests/api-audit/src/test/java/endpoints/dashboard/dashboardReview.feature]: See peerReview.feature comment on apiadmin and the additional statuses not checked here (ToDo)

**Comment by chtompki** [comment]
Do we still want this pull request?

**Comment by amit1s** [comment]
Yes Please

    On Friday, March 15, 2019, 10:48:31 AM PDT, Rob Tompkins <notifications@github.com> wrote:  
 
 
Do we still want this pull request?

**Comment by chtompki** [comment]
Cool. I’ll work towards getting it merged

**Pull request merged**